File: app/src/main/java/com/example/vlad/financemanager/ui/IMoneyCalculation.java
package com.example.vlad.financemanager.ui;

import com.example.vlad.financemanager.data.models.Operation;

import java.util.Date;

public interface IMoneyCalculation {

    int getCategoryId();

    int getAccountId();

    Date getOperationDate();

    String getComment();

    String getAmount();

    boolean getIsOperationInput();

    void setAmountResultText(String resultText);

    void setCalculatorToZero();

    void finishActivity();

    void calculationErrorSignal();

    void calculationErrorSignal(String msg);

    void sendNewOperation(Operation operation);


}

File: app/src/main/java/com/example/vlad/financemanager/ui/IMoneyCalculation.kt
package com.example.vlad.financemanager.ui

import com.example.vlad.financemanager.data.models.Operation

import java.util.Date

interface IMoneyCalculation {

    val categoryId: Int

    val accountId: Int

    val operationDate: Date

    val comment: String

    val amount: String

    val isOperationInput: Boolean

    fun setAmountResultText(resultText: String)

    fun setCalculatorToZero()

    fun finishActivity()

    fun calculationErrorSignal()

    fun calculationErrorSignal(msg: String)

    fun sendNewOperation(operation: Operation)

    fun initUiViaOperationValues(operation: Operation)

}
